### PR TITLE
Bug 2015624: Refactor IBM Cloud skips to External

### PR DESCRIPTION
**Pull request by cjschaef**
Refactor code related to skipping tests on IBM Cloud to skip based
on External ControlPlaneTopology, or Managed offering. IBM Cloud
will be offering an OCP, unmanaged, cluster that should not skip
such tests.

### CODE REVIEW

File: test/extended/authorization/authorization.go [hunk 1]
```diff
@@ -250,16 +252,19 @@ func (test resourceAccessReviewTest) run() {
 				}
 			}
 
+			infra, err := test.ocClient.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (cjschaef) [reply test/extended/authorization/authorization.go]: I hopefully have something sufficient to do that now

File: test/extended/util/framework.go [hunk 2]
```diff
@@ -2002,3 +2006,18 @@ func AllClusterVersionsAreGTE(version semver.Version, config *rest.Config) (bool
 	}
 	return true, nil
 }
+
+// GetControlPlaneTopology retrieves the cluster infrastructure TopologyMode
+func GetControlPlaneTopology(ocClient *CLI) (*configv1.TopologyMode, error) {
+	if ControlPlaneTopology == nil {
```
Comment (alvaroaleman) [inline test/extended/util/framework.go:2012]: I think this needs to be threadsafe/protected by a sync.Mutex as tests could end up calling this in parallel

Comment (cjschaef) [reply test/extended/util/framework.go:2013]: Added a mutex lock to the function

Comment (alvaroaleman) [reply test/extended/util/framework.go:2013]: the mutex already needs to be used for reading ( `if ControlPlaneTopology == nil`) as reading/writing to a variable in parallel is a datarace.

Comment (cjschaef) [reply test/extended/util/framework.go:2013]: Will the additional locking be acceptable, if we always perform it when this function is called, compared with the additional check after receiving the lock, prior to attempting to set the value?

I considered using the builtin `once`, but was not confident that that would work properly if an error was raised while attempting to get the `Infrastructure.Status.ControlPlaneTopology`.

Comment (alvaroaleman) [reply test/extended/util/framework.go:2013]: If we just lock in the very beginning of the function prior to reading we are good. An additional check prior to setting the value would only be needed if we used a RWLock, but I don't think we need that.

Comment (cjschaef) [reply test/extended/util/framework.go:2013]: okay, I'll make that change, thanks!

Comment (cjschaef) [reply test/extended/util/framework.go]: updated

**Review by alvaroaleman — APPROVED** [review]
thanks for your work!

**Pull request merged**